Modules/TestQuestionPool/classes/class.assMatchingQuestion.php

```diff
@@ -318,11 +324,10 @@ public function loadFromDb($question_id): void
             array('integer'),
             array($question_id)
         );
-        include_once "./Modules/TestQuestionPool/classes/class.assAnswerMatchingTerm.php";
-        $this->terms = array();
+        $this->terms = [];
         if ($result->numRows() > 0) {
             while ($data = $ilDB->fetchAssoc($result)) {
-                $term = new assAnswerMatchingTerm($data['term'], $data['picture'], $data['ident']);
+                $term = $this->createMatchingTerm($data['term'], $data['picture'], (int)$data['ident']);
```

Comment:
```suggestion
                $term = $this->createMatchingTerm($data['term'] ?? '', $data['picture'] ?? '', (int) $data['ident']);
```
This can now fail, if $data['term'] === null or $data['picture'] === null